profile/filter_test.go

```diff
@@ -15,60 +15,398 @@
 package profile
 
 import (
+	"fmt"
 	"regexp"
+	"strings"
 	"testing"
+
+	"github.com/google/pprof/internal/proftest"
 )
 
-func TestFilter(t *testing.T) {
-	// Perform several forms of filtering on the test profile.
+var simpleProfileMappings = []*Mapping{
+	{ID: 1, Start: 0x10000, Limit: 0x40000, File: "map0", HasFunctions: true, HasFilenames: true, HasLineNumbers: true, HasInlineFrames: true},
+	{ID: 2, Start: 0x50000, Limit: 0x70000, File: "map1", HasFunctions: true, HasFilenames: true, HasLineNumbers: true, HasInlineFrames: true},
+}
 
-	type filterTestcase struct {
-		focus, ignore, hide, show *regexp.Regexp
-		fm, im, hm, hnm           bool
-	}
+var simpleProfilefunctions = []*Function{
+	{ID: 1, Name: "fun0", SystemName: "fun0", Filename: "file0"},
+	{ID: 2, Name: "fun1", SystemName: "fun1", Filename: "file1"},
+	{ID: 3, Name: "fun2", SystemName: "fun2", Filename: "file2"},
+	{ID: 4, Name: "fun3", SystemName: "fun3", Filename: "file3"},
+	{ID: 5, Name: "fun4", SystemName: "fun4", Filename: "file4"},
+	{ID: 6, Name: "fun5", SystemName: "fun5", Filename: "file5"},
+	{ID: 7, Name: "fun6", SystemName: "fun6", Filename: "file6"},
+	{ID: 8, Name: "fun7", SystemName: "fun7", Filename: "file7"},
+	{ID: 9, Name: "fun8", SystemName: "fun8", Filename: "file8"},
+	{ID: 10, Name: "fun9", SystemName: "fun9", Filename: "file9"},
+	{ID: 11, Name: "fun10", SystemName: "fun10", Filename: "file10"},
+}
 
-	for tx, tc := range []filterTestcase{
+var simpleProfilelocations = []*Location{
```

Comment:
"Simple" is subjective. How about expressing the exact purpose in the name: `noInlinesLocs`?

Reply:
done.